examples/bits.rs

```diff
@@ -1,98 +1,207 @@
-        for byte in &self.bytes {
-            write!(f, "{:0width$b}", byte, width = BITS)?;
-        }
-        Ok(())
-    }
-}
+    // Rust strings can iterate over their individual characters. This block
+    // loops over characters, collecting their start point in the text so that
+    // we can grab the encoded bytes of each one.
+    let mut row_num = 0;
+    for (char_num, (start, codepoint)) in TEXT.char_indices().enumerate() {
+        println!("Character {}: {}", char_num, codepoint);
+        // Each character has a variable width, so we need to find that.
+        let byte_count = codepoint.len_utf8();
+        // And then collect the bytes of the string that make up the character.
+        // `start` gives us the starting position in the text sequence, and
+        // `byte_count` gives us the length in bytes of the character, so we
+        // need to select the range beginning at `start`, running for
+        // `byte_count`. Another style of writing this that you might see in
+        // Rust libraries is `[start ..][.. length]`.
+        let row: &[u8] = &TEXT.as_bytes()[start .. start + byte_count];
+
+        // For each byte (`u8`), we use `bitvec` to make a view into its bits.
+        // `bitvec` provides the `.bits::<_>()` method on Rust integers for easy
+        // access to its view types.
+        //
+        // The `LittleEndian` means that the view moves from least significant
+        // bit to most significant. Since we want to display on screen the most
+        // significant bit on the left, and the least on the right, the turtle
+        // will have to move from right to left to match.
+        //
+        // These types describe different ways to view the same data. You can
+        // read more about them in the `bitvec` docs, and at Wikipedia:
+        // https://docs.rs/bitvec/0.16.1/bitvec/cursor/index.html
+        // https://en.wikipedia.org/wiki/Endianness#Bit_endianness
+        for byte in row {
+            println!("  Byte {:02}:\n    Value: 0x{:02X}\n    Bits: {:08b}", row_num, byte, byte);
+
+            let bits: &BitSlice<_, _> = byte.bits::<LittleEndian>();
 
-impl BitVec {
-    /// Create a new BitVec with the given size in bits
-    /// Note that the actual size of the vector is in bytes so `bits` will be rounded up to the
-    /// nearest multiple of 8 >= bits.
-    pub fn with_capacity(bits: usize) -> Self {
-        if bits > 0 {
-            // Ceiling division of bits / BITS so that we get >= bits capacity
-            // From: https://stackoverflow.com/a/2745086/551904
-            let capacity = 1 + ((bits - 1) / BITS);
-            Self { bytes: vec![0; capacity] }
-        } else {
-            Default::default()
+            // Then we draw the byte's bits as a row
+            draw_row(&mut turtle, bits);
+            // And reset the turtle to the right edge.
+            turtle.set_x(right_edge);
```

Comment:
What do you think of changing this to use `left`/`right` and `forward`? `set_x` is convenient, and your use of `go_to` in the initialization code above is fine, but using it in the middle of an animation usually isn't recommended. Methods like `set_x` and `go_to` make the turtle jump directly to the given position instead of doing its usual animated movement. 

It also makes the code a little less extensible. For example, you can't just add a rotation at the top to make the entire drawing rotate.

<details><summary>More details about that last sentence</summary>
<p>

Consider the following program to draw a square:

```rust
use turtle::Turtle;

fn main() {
    let mut turtle = Turtle::new();

    for _ in 0..4 {
        turtle.forward(200.0);
        turtle.right(90.0);
    }
}
```

One of the best (and sometimes worst) features of turtle is that you can rotate the entire drawing just by adding a call to `right` in the beginning. The rest of the code stays the same:


```rust
use turtle::Turtle;

fn main() {
    let mut turtle = Turtle::new();

    // Rotates the entire drawing
    turtle.right(30.0);

    for _ in 0..4 {
        turtle.forward(200.0);
        turtle.right(90.0);
    }
}
```

This is because by default, everything in turtle is done relative to its current position and orientation. The `set_x`, `go_to`, etc. methods break that functionality. Using those methods builds in an assumption that the drawing is always going to look or be oriented a certain way. It's fine to use them in certain cases (e.g. in the initialization above), but in general it's good to stay away from them in the middle of a drawing if you can.

</p>
</details>

There are other uses of `set_x`, `set_heading`, `go_to`, etc. that you may want to consider changing if you decide to change this one.

Reply:
Awkward restructuring to make this work but ultimately worth it